pypeit/core/findobj_skymask.py

```diff
@@ -147,7 +147,1159 @@ def create_skymask(sobjs, thismask, slit_left, slit_righ, box_rad_pix=None, trim
+
+        trim_edg (:obj:`tuple`, optional):
+            A two-tuple of integers or floats used to ignore objects within this
+            many pixels of the left and right slit boundaries, respectively.
+        show_peaks (:obj:`bool`, optional):
+            Plot the QA of the object peak finding in each order.
+        show_fits (:obj:`bool`, optional):
+            Plot trace fitting for final fits using PCA as crutch.
+        show_single_fits (:obj:`bool`, optional):
+            Plot trace fitting for single order fits.
+        show_trace (:obj:`bool`, optional):
+            Display the object traces on top of the image.
+        show_single_trace (:obj:`bool`, optional):
+            Display the object traces on top of the single order.
+        show_pca (:obj:`bool`, optional):
+            Display debugging plots for the PCA decomposition.
+        debug_all (:obj:`bool`, optional):
+            Show all the debugging plots.  If True, this also overrides any
+            provided values for ``show_peaks``, ``show_trace``, and
+            ``show_pca``, setting them to True.
+        objfindQA_filename (:obj:`str`, optional):
+            Full path (directory and filename) for the object profile QA plot.
+            If None, not plot is produced and saved.
+        manual_extract_dict : :obj:`dict`, optional
+            Dict guiding the manual extraction
+
+    Returns:
+        :class:`~pypeit.specobjs.SpecObjs`: Object containing the objects
+        detected.
+    """
+        #debug_all=True
+    if debug_all:
+        show_peaks = True
+        #show_fits = True
+        #show_single_fits = True
+        show_trace = True
+        show_pca = True
+        #show_single_trace = True
+
+    if specobj_dict is None:
+        specobj_dict = {'SLITID': 999, 
+                        'ECH_ORDERINDX': 999,
+                        'DET': det, 'OBJTYPE': 'unknown', 
+                        'PYPELINE': 'Echelle'}
+
+    # Loop over the orders and find the objects within them (if any)
+    order_gpm = np.invert(slits_bpm)
+    sobjs_in_orders = ech_findobj_ineach_order(
+        image, ivar, slitmask, slit_left, 
+        slit_righ, slit_spat_id,
+        order_vec, order_gpm,
+        spec_min_max, plate_scale,
+        det=det,
+        inmask=inmask, 
+        std_trace=std_trace,
+        specobj_dict=specobj_dict,
+        snr_thresh=snr_thresh,
+        show_peaks=show_peaks, 
+        show_single_fits=show_single_fits,
+        show_single_trace=show_single_trace,
+        extract_maskwidth=extract_maskwidth,
+        trim_edg=trim_edg,
+        fwhm=fwhm,
+        use_user_fwhm=use_user_fwhm,
+        nperorder=nperorder,
+        maxdev=maxdev,
+        box_radius=box_radius,
+        objfindQA_filename=objfindQA_filename,
+        hand_extract_dict=manual_extract_dict)
+
+    # Lots of additional work for slits with sources (found or input manually)
+    if len(sobjs_in_orders) > 0: # or manual_extract_dict is not None:
```

Comment:
Instead of this `if..else` block, I would immediately return the `else` case so that the rest of it doesn't need to be within an if statement.  I.e.:
```
if len(sobjs_in_orders) == 0:
    return sobjs_in_orders

...

return ech_pca_traces(...)
```

Reply:
good call